### PR TITLE
added the first thai classification dataset

### PR DESCRIPTION
**Pull request by Akash190104**
Wisesight Sentiment Corpus: Social media messages in Thai language with sentiment label (positive, neutral, negative, question)
## Checklist for adding MMTEB dataset

Reason for dataset addition:

This is the first classification dataset for Thai.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/tha/wisesight_sentiment_classification.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        name="wisesight_sentiment",
+        description="Wisesight Sentiment Corpus: Social media messages in Thai language with sentiment label (positive, neutral, negative, question)",
+        reference="https://github.com/PyThaiNLP/wisesight-sentiment",
+        dataset={
+            "path": "wisesight_sentiment",
+            "revision": "14aa5773afa135ba835cc5179bbc4a63657a42ae",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train", "test", "validation"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/tha/wisesight_sentiment_classification.py:18]: Seems like you can go for the test set.

File: mteb/tasks/Classification/tha/wisesight_sentiment_classification.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    )
+
+    def dataset_transform(self):
+        N_SAMPLES = 2048
+        TEST_SAMPLES = 1024  # define this as per your requirement
+
+        # Transform train set
+        self.dataset["train"] = self.dataset["train"].select(range(N_SAMPLES))
+        self.dataset["train"] = self.dataset["train"].rename_column("texts", "text")
+        self.dataset["train"] = self.dataset["train"].rename_column("category", "label")
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/tha/wisesight_sentiment_classification.py:56]: I would do the downsample using the suggested (`self.stratified_subsampling()`) so ensure that we stratify based on the labels.

**Review by KennethEnevoldsen — COMMENTED** [review]
A few minor comments; otherwise, everything looks great. Feel free to add points as well.

File: mteb/tasks/Classification/tha/wisesight_sentiment_classification.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  month        = sep,
+  year         = 2019,
+  publisher    = {Zenodo},
+  version      = {v1.0},
+  doi          = {10.5281/zenodo.3457447},
+  url          = {https://doi.org/10.5281/zenodo.3457447}
+}
+
+""",
+        n_samples={"train": 3418},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/tha/wisesight_sentiment_classification.py:45]:
```suggestion
        n_samples={"train": 2048},
```

File: docs/mmteb/points/538.jsonl [hunk 4]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "Akash190104", "New dataset": 6}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/538.jsonl:1]:
```suggestion
{"GitHub": "Akash190104", "New dataset": 2}
```
thai already included in MTOPIntentClassification

**Comment by KennethEnevoldsen** [comment]
Looks great @Akash190104, I changed the points to 2 (tha is included in MTOPIntentClassification). Let me know if that is wrong.

**Comment by Akash190104** [comment]
> Looks great @Akash190104, I changed the points to 2 (tha is included in MTOPIntentClassification). Let me know if that is wrong.

Yeah thank makes sense.

**Pull request merged**